File: pi/C++/piserial.cc
#include <cstdlib>
#include <cmath>
#include <iostream>

using namespace std;

#define N 840

int main(void)
{
  double pi, exactpi;

  cout << "Computing approximation to pi using N = " << N << endl;

  pi = 0.0;

  for (int i=1; i<=N; i++)
    {
      pi = pi + 1.0/( 1.0 + pow( (((double) i)-0.5)/((double) N), 2.0) );
    }

  pi = pi * 4.0/((double) N);

  exactpi = 4.0*atan(1.0);

  cout << "pi = " << pi << ", % error = "
       << fabs(100.0*(pi-exactpi)/exactpi) << endl;

  return 0;
}

File: pi/P/piserial.py
#!/usr/bin/env python

import sys
import math

def main(argv):

    N = 840

    print(f"Computing approximation to pi using N = {N}")

    pi = 0.0

    for i in range(1, N+1):
        pi = pi + 1 / ( 1 + ((i-1/2)/N)**2)

    pi = pi * 4/N

    exactpi = 4.0*math.atan(1.0)
  
    print(f"pi = {pi}, % error = {math.fabs(100.0*(pi-exactpi)/exactpi)}")


if __name__ == "__main__":
    main(sys.argv[1:])
